// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/actions/shared/messageProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMessageProperties = getMessageProperties;
const Message_1 = require("../../Message");
function getMessageProperties(operations) {
    return {
        displayName: 'Messages',
        name: 'messages',
        type: 'fixedCollection',
        typeOptions: {
            multipleValues: true,
            sortable: true,
        },
        default: {},
        placeholder: 'Add Message',
        displayOptions: {
            show: {
                operation: operations,
            },
        },
        options: [
            {
                displayName: 'Message',
                name: 'values',
                values: [
                    {
                        displayName: 'Alt Text',
                        name: 'altText',
                        type: 'string',
                        required: true,
                        default: '',
                        placeholder: 'This is a flex message',
                        description: 'Alternative text shown in notifications and when Flex message cannot be displayed',
                        displayOptions: {
                            show: {
                                type: [Message_1.MessageType.Flex],
                            },
                        },
                    },
                    {
                        displayName: 'Flex Message JSON',
                        name: 'flexJson',
                        type: 'json',
                        typeOptions: {
                            rows: 10,
                        },
                        default: '{\n  "type": "bubble",\n  "body": {\n    "type": "box",\n    "layout": "vertical",\n    "contents": [\n      {\n        "type": "text",\n        "text": "Hello, World!"\n      }\n    ]\n  }\n}',
                        description: 'Flex Message container JSON. Design your message using <a href="https://developers.line.biz/flex-simulator/" target="_blank">LINE Flex Message Simulator</a>.',
                        displayOptions: {
                            show: {
                                type: [Message_1.MessageType.Flex],
                            },
                        },
                    },
                    {
                        displayName: 'Quote Token',
                        name: 'quoteToken',
                        type: 'string',
                        typeOptions: { password: true },
                        default: '',
                        placeholder: '1234567890',
                        description: 'The quote token to use for quoting a message. If not provided, the message will be sent as a new message.',
                        displayOptions: {
                            show: {
                                type: [Message_1.MessageType.TextV2],
                            },
                        },
                    },
                    {
                        displayName: 'Text',
                        name: 'text',
                        type: 'string',
                        default: '',
                        placeholder: 'Hello, World!',
                        description: 'The text message to send',
                        displayOptions: {
                            show: {
                                type: [Message_1.MessageType.TextV2],
                            },
                        },
                    },
                    {
                        displayName: 'Type',
                        name: 'type',
                        type: 'options',
                        default: 'textV2',
                        options: [
                            {
                                name: 'Flex Message',
                                value: Message_1.MessageType.Flex,
                            },
                            {
                                name: 'Text Message (V2)',
                                value: Message_1.MessageType.TextV2,
                            },
                        ],
                    },
                    {
                        displayName: 'Quick Reply (Optional)',
                        name: 'quickReply',
                        type: 'fixedCollection',
                        typeOptions: {
                            multipleValues: true,
                            sortable: true,
                        },
                        default: {},
                        placeholder: 'Add Quick Reply',
                        options: [
                            {
                                displayName: 'Quick Reply Item',
                                name: 'items',
                                values: [
                                    {
                                        displayName: 'Action Type',
                                        name: 'actionType',
                                        type: 'options',
                                        required: true,
                                        default: 'message',
                                        options: [
                                            {
                                                name: 'Message',
                                                value: Message_1.ActionType.Message,
                                            },
                                            {
                                                name: 'Postback',
                                                value: Message_1.ActionType.Postback,
                                            },
                                        ],
                                    },
                                    {
                                        displayName: 'Data',
                                        name: 'data',
                                        type: 'string',
                                        default: '',
                                        placeholder: '{"key":"value"}',
                                        description: 'The data to send with the quick reply action. Format depends on action type.',
                                    },
                                    {
                                        displayName: 'Label',
                                        name: 'label',
                                        type: 'string',
                                        default: '',
                                        placeholder: 'Quick Reply',
                                        description: 'The label for the quick reply item',
                                    },
                                ],
                            },
                        ],
                    },
                ],
            },
        ],
    };
}
