// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/actions/shared/operationProperties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getReplyTokenProperty = getReplyTokenProperty;
exports.getUserIdProperty = getUserIdProperty;
function getReplyTokenProperty() {
    return {
        displayName: 'Reply Token',
        name: 'replyToken',
        type: 'string',
        required: true,
        typeOptions: { password: true },
        default: '',
        placeholder: '1234567890',
        description: 'The reply token to use for replying to a message. If not provided, the message will be sent as a new message.',
        displayOptions: {
            show: {
                operation: ['reply'],
            },
        },
    };
}
function getUserIdProperty() {
    return {
        displayName: 'User ID',
        name: 'to',
        type: 'string',
        required: true,
        default: '',
        placeholder: 'U1234567890abcdef...',
        description: 'The ID of the user to send the message to',
        displayOptions: {
            show: {
                operation: ['send'],
            },
        },
    };
}
