// @aotoki/n8n-nodes-line-messaging/dist/credentials/LineMessagingApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineMessagingApi = void 0;
class LineMessagingApi {
    constructor() {
        this.name = 'lineMessagingApi';
        this.displayName = 'Line Messaging API';
        this.documentationUrl = 'https://developers.line.biz/en/docs/messaging-api/overview/';
        this.icon = 'file:../nodes/LineMessaging/line.svg';
        this.properties = [
            {
                displayName: 'Channel Access Token',
                name: 'accessToken',
                type: 'string',
                typeOptions: { password: true },
                default: '',
                description: 'The Channel Access Token for the Line Messaging API. You can obtain it from the Line Developers Console.',
            },
            {
                displayName: 'Channel Secret',
                name: 'channelSecret',
                type: 'string',
                typeOptions: { password: true },
                default: '',
                description: 'The Channel Secret for the Line Messaging API. You can obtain it from the Line Developers Console.',
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                headers: {
                    Authorization: '={{`Bearer ${$credentials.accessToken}`}}',
                },
            },
        };
        this.test = {
            request: {
                method: 'GET',
                baseURL: 'https://api.line.me/v2/bot',
                url: '/info',
                json: true,
            },
        };
    }
}
exports.LineMessagingApi = LineMessagingApi;
