// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/actions/getProfile.operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = exports.properties = void 0;
exports.execute = execute;
const n8n_workflow_1 = require("n8n-workflow");
const GenericFunctions_1 = require("../GenericFunctions");
exports.properties = [
    {
        displayName: 'User ID',
        name: 'userId',
        type: 'string',
        default: '',
        required: true,
        placeholder: 'U4af4980629...',
        description: 'The ID of the user whose profile you want to retrieve',
        displayOptions: {
            show: {
                operation: ['getProfile'],
            },
        },
    },
];
exports.description = exports.properties;
async function execute(items) {
    const returnData = [];
    for (let i = 0; i < items.length; i++) {
        try {
            const userId = this.getNodeParameter('userId', i);
            if (!userId) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'User ID is required. Use the LINE user ID from webhook or other sources.');
            }
            const responseData = await GenericFunctions_1.apiRequest.call(this, 'GET', `/profile/${userId}`, {});
            returnData.push({
                json: {
                    success: true,
                    profile: responseData,
                },
                pairedItem: {
                    item: i,
                },
            });
        }
        catch (error) {
            if (this.continueOnFail()) {
                returnData.push({
                    json: {
                        success: false,
                        error: error.message,
                    },
                    pairedItem: {
                        item: i,
                    },
                });
                continue;
            }
            throw error;
        }
    }
    return returnData;
}
