// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/actions/multicast.operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = exports.properties = void 0;
exports.execute = execute;
const n8n_workflow_1 = require("n8n-workflow");
const GenericFunctions_1 = require("../GenericFunctions");
const Factory_1 = require("../Factory");
const messageProperties_1 = require("./shared/messageProperties");
function getUserIdsProperty() {
    return {
        displayName: 'User IDs',
        name: 'to',
        type: 'string',
        required: true,
        default: '',
        placeholder: 'U1234567890abcdef..., U0987654321fedcba...',
        description: 'Comma-separated list of User IDs to send the message to (max 500 recipients)',
        displayOptions: {
            show: {
                operation: ['multicast'],
            },
        },
    };
}
exports.properties = [
    getUserIdsProperty(),
    (0, messageProperties_1.getMessageProperties)(['multicast']),
];
exports.description = exports.properties;
async function execute(items) {
    const returnData = [];
    for (let i = 0; i < items.length; i++) {
        try {
            const messagesCollection = this.getNodeParameter('messages', i, {
                values: [],
            });
            const messages = (messagesCollection.values || []).map((params) => Factory_1.MessageFactory.createMessage(params));
            const toInput = this.getNodeParameter('to', i, '');
            if (!toInput) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'User IDs are required. Provide comma-separated LINE user IDs.');
            }
            const to = toInput
                .split(',')
                .map((id) => id.trim())
                .filter((id) => id.length > 0);
            if (to.length === 0) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'At least one valid User ID is required. Check your comma-separated list.');
            }
            if (to.length > 500) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'Maximum 500 recipients allowed. You provided ' + to.length + ' user IDs.');
            }
            if (messages.length === 0) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'At least one message is required. Add text, image, or other message types.');
            }
            const responseData = await GenericFunctions_1.apiRequest.call(this, 'POST', '/message/multicast', {
                to,
                messages,
            });
            returnData.push({
                json: {
                    success: true,
                    response: responseData,
                    recipientCount: to.length,
                    userIds: to,
                },
                pairedItem: {
                    item: i,
                },
            });
        }
        catch (error) {
            if (this.continueOnFail()) {
                returnData.push({
                    json: {
                        success: false,
                        error: error.message,
                    },
                    pairedItem: {
                        item: i,
                    },
                });
                continue;
            }
            throw error;
        }
    }
    return returnData;
}
