// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/actions/send.operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = exports.properties = void 0;
exports.execute = execute;
const n8n_workflow_1 = require("n8n-workflow");
const GenericFunctions_1 = require("../GenericFunctions");
const Factory_1 = require("../Factory");
const operationProperties_1 = require("./shared/operationProperties");
const messageProperties_1 = require("./shared/messageProperties");
exports.properties = [
    (0, operationProperties_1.getUserIdProperty)(),
    (0, messageProperties_1.getMessageProperties)(['send']),
];
exports.description = exports.properties;
async function execute(items) {
    const returnData = [];
    for (let i = 0; i < items.length; i++) {
        try {
            const messagesCollection = this.getNodeParameter('messages', i, {
                values: [],
            });
            const messages = (messagesCollection.values || []).map((params) => Factory_1.MessageFactory.createMessage(params));
            const to = this.getNodeParameter('to', i, '');
            if (!to) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'User ID is required. Use the LINE user ID from webhook or profile.');
            }
            if (messages.length === 0) {
                throw new n8n_workflow_1.NodeOperationError(this.getNode(), 'At least one message is required. Add text, image, or other message types.');
            }
            const responseData = await GenericFunctions_1.apiRequest.call(this, 'POST', '/message/push', {
                to,
                messages,
            });
            returnData.push({
                json: { success: true, response: responseData },
                pairedItem: {
                    item: i,
                },
            });
        }
        catch (error) {
            if (this.continueOnFail()) {
                returnData.push({
                    json: {
                        success: false,
                        error: error.message,
                    },
                    pairedItem: {
                        item: i,
                    },
                });
                continue;
            }
            throw error;
        }
    }
    return returnData;
}
