// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/LineMessagingData.node.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineMessagingData = void 0;
const getContent = __importStar(require("./data-actions/getContent.operation"));
class LineMessagingData {
    constructor() {
        this.description = {
            displayName: 'Line Messaging Data',
            name: 'lineMessagingData',
            icon: 'file:line.svg',
            group: ['output'],
            version: [1],
            subtitle: '={{$parameter["operation"]}}',
            description: 'Send and receive large amounts of data with Line Messaging API',
            defaults: {
                name: 'Line Messaging Data',
            },
            usableAsTool: true,
            inputs: ["main"],
            outputs: ["main"],
            credentials: [
                {
                    name: 'lineMessagingApi',
                    required: true,
                },
            ],
            properties: [
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    default: 'getContent',
                    options: [
                        {
                            name: 'Get Content',
                            value: 'getContent',
                            description: 'Retrieve content of a message by its ID',
                            action: 'Retrieve content of a message by its ID',
                        },
                    ],
                },
                ...getContent.description,
            ],
        };
    }
    async execute() {
        const items = this.getInputData();
        const operation = this.getNodeParameter('operation', 0);
        let returnData = [];
        if (operation === 'getContent') {
            returnData = await getContent.execute.call(this, items);
        }
        return [returnData];
    }
}
exports.LineMessagingData = LineMessagingData;
