// @aotoki/n8n-nodes-line-messaging/dist/nodes/LineMessaging/LineMessaging.node.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineMessaging = void 0;
const reply = __importStar(require("./actions/reply.operation"));
const getProfile = __importStar(require("./actions/getProfile.operation"));
const send = __importStar(require("./actions/send.operation"));
const multicast = __importStar(require("./actions/multicast.operation"));
const showLoadingAnimation = __importStar(require("./actions/showLoadingAnimation.operation"));
class LineMessaging {
    constructor() {
        this.description = {
            displayName: 'Line Messaging',
            name: 'lineMessaging',
            icon: 'file:line.svg',
            group: ['output'],
            version: [1],
            subtitle: '={{$parameter["operation"]}}',
            description: 'Interact with the Line Messaging API',
            defaults: {
                name: 'Line Messaging',
            },
            usableAsTool: true,
            inputs: ["main"],
            outputs: ["main"],
            credentials: [
                {
                    name: 'lineMessagingApi',
                    required: true,
                },
            ],
            properties: [
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    default: 'reply',
                    noDataExpression: true,
                    options: [
                        {
                            name: 'Get Profile',
                            value: 'getProfile',
                            description: 'Retrieve profile information of a user',
                            action: 'Retrieve profile information of a user',
                        },
                        {
                            name: 'Multicast',
                            value: 'multicast',
                            description: 'Send a message to multiple users',
                            action: 'Send a message to multiple users',
                        },
                        {
                            name: 'Reply',
                            value: 'reply',
                            description: 'Reply to a message using a reply token',
                            action: 'Reply to a message using a reply token',
                        },
                        {
                            name: 'Send',
                            value: 'send',
                            description: 'Send a message to a user',
                            action: 'Send a message to a user',
                        },
                        {
                            name: 'Show Loading Animation',
                            value: 'showLoadingAnimation',
                            description: 'Display a loading animation to a user',
                            action: 'Display a loading animation to a user',
                        },
                    ],
                },
                ...reply.description,
                ...send.description,
                ...multicast.description,
                ...getProfile.description,
                ...showLoadingAnimation.description,
            ],
        };
    }
    async execute() {
        const items = this.getInputData();
        const operation = this.getNodeParameter('operation', 0);
        let returnData = [];
        if (operation === 'reply') {
            returnData = await reply.execute.call(this, items);
        }
        else if (operation === 'send') {
            returnData = await send.execute.call(this, items);
        }
        else if (operation === 'multicast') {
            returnData = await multicast.execute.call(this, items);
        }
        else if (operation === 'getProfile') {
            returnData = await getProfile.execute.call(this, items);
        }
        else if (operation === 'showLoadingAnimation') {
            returnData = await showLoadingAnimation.execute.call(this, items);
        }
        return [returnData];
    }
}
exports.LineMessaging = LineMessaging;
